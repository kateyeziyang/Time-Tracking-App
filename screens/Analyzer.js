import * as React from 'react';
import { useState } from 'react';
import { Text, View, Dimensions, TextInput, StyleSheet } from 'react-native';
import { Button } from 'react-native-elements';
import { Picker } from '@react-native-community/picker';
import AsyncStorage from '@react-native-community/async-storage';
import { submitPeriod, summarizeDataByTag, summarizeDataByTagAndDate, summarizeData } from '../common/commonFunctions';

import {
  PieChart,
  StackedBarChart
} from "react-native-chart-kit";

/**
 * Show statistics based on past record.
 * @return A page of stored data
*/
export function AnalyzerScreen({ extra }) {
  const chartConfig = {
  backgroundGradientFrom: "#1E2923",
  backgroundGradientFromOpacity: 0,
  backgroundGradientTo: "#08130D",
  backgroundGradientToOpacity: 0.5,
  color: (opacity = 1) => `rgba(26, 255, 146, ${opacity})`,
  strokeWidth: 2, // optional, default 3
  barPercentage: 0.5
  };

  const [value, onChangeText] = React.useState('04/13/20');
  const [value2, onChangeText2] = React.useState('04/19/20');
  const [selectedValue, setSelectedValue] = useState("Pie");
  const [chartData, setChartData] = useState([]);

  if (extra) {
    console.log("Test condition.");
    const [chartData, setChartData] = useState(extra);
  } else {
    const [chartData, setChartData] = useState([]);
  }

  let chartImage;
  let textAnalysis;
  if (chartData.length !== 0) {
    let sumData = summarizeData(chartData);
    let numDays = Object.keys(chartData).length;
    let averageList = [sumData["Study"], sumData["Work"], sumData["Relax"], sumData["Sleep"]];
    for (let i = 0; i < averageList.length; i++) {
      averageList[i] = averageList[i] / numDays;
    }
    textAnalysis =
    <View style={{padding:10}}>
        <Text style={styles.bigText}>Average time spending per day (in mins)</Text>
        <View style={{flexDirection: 'row', paddingTop: 10, alignItems: 'center'}}>
          <Button title="Study" buttonStyle={styles.buttonStyle1} />
          <View style={{width:5}}>
          </View>
          <Text style={styles.bigText}>{averageList[0].toFixed()}</Text>
          <View style={{width:5}}>
          </View>
          <Button title="Work" buttonStyle={styles.buttonStyle2} />
          <View style={{width:5}}>
          </View>
          <Text style={styles.bigText}>{averageList[1].toFixed()}</Text>
          <View style={{width:5}}>
          </View>
          <Button title="Relax" buttonStyle={styles.buttonStyle3} />
          <View style={{width:5}}>
          </View>
          <Text style={styles.bigText}>{averageList[2].toFixed()}</Text>
          <View style={{width:5}}>
          </View>
          <Button title="Sleep" buttonStyle={styles.buttonStyle4} />
          <View style={{width:5}}>
          </View>
          <Text style={styles.bigText}>{averageList[3].toFixed()}</Text>
        </View>
      </View>
    ;
    switch (selectedValue) {
      case "Pie":
        let tagsCount = summarizeDataByTag(chartData);
        chartImage = <PieChart
                      data={tagsCount}
                      width={350}
                      height={220}
                      chartConfig={chartConfig}
                      accessor="count"
                      backgroundColor="transparent"
                      paddingLeft="15"
                    />
        break;
      case "Stacked":
        let barData = summarizeDataByTagAndDate(chartData);
        console.log(barData);
        chartImage = <StackedBarChart
                      data={barData}
                      width={350}
                      height={220}
                      chartConfig={{
                      backgroundColor: '#1cc910',
                      backgroundGradientFrom: '#eff3ff',
                      backgroundGradientTo: '#efefef',
                      decimalPlaces: 0,
                      color: (opacity = 1) => `rgba(0, 0, 0, ${opacity})`,
                      style: {
                        borderRadius: 16,
                      },
                      }}
                      style={{
                        marginVertical: 8,
                        borderRadius: 50,
                      }}
                      />
        break;
      default:
        chartImage = null;
    }
  } else {
    chartImage = null;
    textAnalysis = null;
  }

  return (
    <View style={{flex: 1, paddingLeft: 20, paddingRight: 20}}>
      <View style={{padding:10}}>
        <View style={{flexDirection: 'row', height: 30}}>
          <View style={{width: 20, height: 30, justifyContent: 'center'}} >
            <Text style={styles.circled}>❶</Text>
          </View>
          <View style={{width: 10, height: 30}} >
          </View>
          <View style={{width: 200, height: 30, justifyContent: 'center'}} >
            <Text style={styles.bigText}>Pick a Time Range</Text>
          </View>
        </View>
      </View>
      <View style={{flexDirection: 'row', justifyContent: 'center'}}>
        <View style={{width: 50, alignItems: 'center', justifyContent: 'center'}} >
        <Text>From</Text>
        </View>
        <View style={{width: 70}} >
          <TextInput
            style={{ height: 40, borderColor: '#c2e8ce', borderWidth: 2, backgroundColor: 'white' }}
            onChangeText={text => onChangeText(text)}
            value={value}
          />
        </View>
        <View style={{width: 30, alignItems: 'center', justifyContent: 'center'}} >
          <Text>To</Text>
        </View>
        <View style={{width: 70}} >
          <TextInput
            style={{ height: 40, borderColor: '#c2e8ce', borderWidth: 2, backgroundColor: 'white'}}
            onChangeText={text => onChangeText2(text)}
            value={value2}
          />
        </View>
        <View style={{width: 90, alignItems: 'center', justifyContent: 'center'}} >
          <Text style={styles.smallText}>MM/DD/YY</Text>
        </View>
      </View>
      <View style={{padding:10}}>
        <View style={{flexDirection: 'row', height: 30}}>
          <View style={{width: 20, height: 30, justifyContent: 'center'}} >
            <Text style={styles.circled}>❷</Text>
          </View>
          <View style={{width: 10, height: 30}} >
          </View>
          <View style={{width: 200, height: 30, justifyContent: 'center'}} >
            <Text style={styles.bigText}>Select Chart Type</Text>
          </View>
        </View>
        <View style={{flexDirection: 'row', height: 40, justifyContent: 'center', alignItems: 'center'}}>
          <View style={{ height: 50, width: 190}}>
            <Picker
              selectedValue={selectedValue}
              style={{ height: 50, width: 200}}
              onValueChange={(itemValue, itemIndex) => setSelectedValue(itemValue)}
            >
              <Picker.Item label="Pie Chart" value="Pie" />
              <Picker.Item label="Stacked-Bar Chart" value="Stacked" />
            </Picker>
          </View>
          <View style={{ width: 20}}>
          </View>
          <View style={{ width: 50, height: 30}}>
            <Button
            raised={true}
            onPress={() => {
              submitPeriod(value, value2, setChartData);
            }}
            title="OK" buttonStyle={styles.buttonStyle}
            />
          </View>
        </View>
      </View>
      {textAnalysis}
      {chartImage}
    </View>
  );
}

const styles = StyleSheet.create({
  bigText: {
    fontSize: 18,
  },
  smallText: {
    fontSize: 14,
    fontStyle: "italic"
  },
  circled: {
    fontSize: 25,
    color: '#5eb7b7'
  },
  buttonStyle: {
    borderRadius: 14,
    backgroundColor: '#fc7978'
  },
  buttonStyle1: {
    borderRadius: 14,
    backgroundColor: '#363062'
  },
  buttonStyle2: {
    borderRadius: 14,
    backgroundColor: '#4d4c7d'
  },
  buttonStyle3: {
    borderRadius: 14,
    backgroundColor: '#827397'
  },
  buttonStyle4: {
    borderRadius: 14,
    backgroundColor: '#d8b9c3'
  },
});
