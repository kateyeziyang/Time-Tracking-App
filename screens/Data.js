import * as React from 'react';
import { useState } from 'react';
import { Text, View, ScrollView, StyleSheet } from 'react-native';
import { Button } from 'react-native-elements';
import AsyncStorage from '@react-native-community/async-storage';
import { submitRecord, clearAll, getAllItemsWrapper } from '../common/commonFunctions';
import { isValidDate } from '../common/commonFunctions';

/**
 * Read stored data and render it in human-readable format.
 * @return A page of stored data
*/
export function DataScreen({ extra }) {
  const RecordView = ({recordData, recordKey}) => {
    return (
      <Text key={recordKey}>
      {recordData}
      </Text>
    );
  };

  const ItemView = ({extraKey}) => {

    let itemData = extra[extraKey];
    let itemIndex = -1;
    let tempStart, tempEnd;

    let itemListArr = itemData.map(itemDat => {
      itemIndex++;

      if (isValidDate(itemDat.start) && isValidDate(itemDat.end)) {
        tempStart = new Date(itemDat.start);
        tempEnd = new Date(itemDat.end);
        return (
          <View key={extraKey+itemIndex+"-View"}
          style={{justifyContent: 'center', padding: 15, borderBottomColor: 'grey', borderBottomWidth: 1}}
          >
            <Text key={extraKey+itemIndex}>
            Date: {extraKey}{'\n'}
            Start: {tempStart.toLocaleString()}{'\n'}
            End: {tempEnd.toLocaleString()}{'\n'}
            Tag: {itemDat.tag}
            </Text>
          </View>
        )
      } else return(null);
    });
    return (itemListArr);
  };

  let textListArr;
  if (extra) {
    let extraKeys = Object.keys(extra);
    textListArr = extraKeys.map(extraKey => {
        return(
          <ItemView key={extraKey+"-ItemView"} extraKey={extraKey}/>
        )
    });
  } else {
    return(null);
  }

  const [data, setData] = useState(textListArr);

  return (
    <View style={{flex: 1}}>
      <View elevation={5} style={{flex:0.08, flexDirection: 'row', justifyContent: 'center',
        padding: 15, backgroundColor: '#424874'}}>
        <View style={{width: 80, height: 40}} >
          <Button onPress={()=>{ clearAll(); setData(null); }} title="Clear" type="outline"
          buttonStyle={styles.buttonStyle} titleStyle={styles.buttonTitleStyle}/>
        </View>
        <View style={{width: 220, justifyContent: 'center', alignItems: 'center'}} >
          <Text style={styles.screenTitle}>
          D A T A
          </Text>
        </View>
        <View style={{width: 80, height: 40}} >
          <Button onPress={()=>{ getAllItemsWrapper(setData); }} title="Refresh"
          buttonStyle={styles.buttonStyle2} type="outline"/>
        </View>
      </View>
      <ScrollView style={{flex: 0.5, paddingLeft: 20, paddingRight: 20, paddingBottom: 40}}>
      {data}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  screenTitle: {
    fontSize: 25,
    color: 'white',
  },
  buttonStyle: {
    borderWidth: 0.2,
    backgroundColor: 'red',
    borderColor: 'white',
  },
  buttonStyle2: {
    borderWidth: 0,
    borderColor: 'black',
    backgroundColor: 'white',
  },
  buttonTitleStyle: {
    color: 'white',
  },
});
