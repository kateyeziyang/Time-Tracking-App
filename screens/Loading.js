import * as React from 'react';
import { Text, View } from 'react-native';

/**
 * Default screen when github haven't return anything.
 * @return A waiting page
*/
export function LoadingScreen() {
  return (
    <View style={{flex:1, justifyContent: 'center', alignItems: 'center'}}>
      <Text>Loading...</Text>
    </View>
  );
}
