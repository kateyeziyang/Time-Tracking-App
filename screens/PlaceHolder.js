import * as React from 'react';
import { Text, View } from 'react-native';

/**
 * Default screen when github haven't return anything.
 * @return A waiting page
*/
export function PlaceHolderScreen() {
  return (
    <View style={{flex:1, justifyContent: 'center', alignItems: 'center'}}>
      <Text>PlaceHolder...</Text>
    </View>
  );
}
