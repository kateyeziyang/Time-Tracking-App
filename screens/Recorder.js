import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet, Platform, } from 'react-native';
import { Button } from 'react-native-elements';
import DateTimePicker from '@react-native-community/datetimepicker';
import { Picker } from '@react-native-community/picker';
import { submitRecord } from '../common/commonFunctions';

/**
 * Recorder screen where user will be able to customize a time period and submit.
 * @return a time recorder page.
*/
export function RecorderScreen({ curDate }) {
  /* ---time picker variables--- */
  const [startDate, setStartDate] = useState(curDate);
  const [endDate, setEndDate] = useState(curDate);
  const [mode, setMode] = useState('date');
  const [show, setShow] = useState(false);
  const [mode2, setMode2] = useState('date');
  const [show2, setShow2] = useState(false);

  const onStartChange = (event, selectedDate) => {
    const currentDate = selectedDate || startDate;
    setShow(Platform.OS === 'ios');
    setStartDate(currentDate);
  };

  const onEndChange = (event, selectedDate) => {
    const currentDate = selectedDate || endDate;
    setShow2(Platform.OS === 'ios');
    setEndDate(currentDate);
  };

  const showMode = currentMode => {
    setShow(true);
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode('date');
  };

  const showTimepicker = () => {
    showMode('time');
  };

  const showMode2 = currentMode => {
    setShow2(true);
    setMode2(currentMode);
  };

  const showDatepicker2 = () => {
    showMode2('date');
  };

  const showTimepicker2 = () => {
    showMode2('time');
  };

  // If set button title directy with the raw string, the newline character won't rendered correctly
  const endButtonTitle1 = "End\nDate\npicker";
  const endButtonTitle2 = "End\ntime\npicker";
  /* ---time picker variables end--- */

  /* ---Event tags variables--- */
  const [selectedValue, setSelectedValue] = useState("N/A");
  /* ---Event tags variables end--- */

  /* ---Custom variables--- */
  /* ---Custom variables end--- */

  return (
    <View style={{flex: 1}}>
      <View style={{padding: 20}}>
        <Text style={styles.bigOrange1}>
        Start time
        </Text>
        <Text style={styles.timeText}>
        {startDate.toDateString()}, {startDate.toLocaleTimeString()}
        </Text>
        <View style={{flexDirection: 'row', justifyContent: 'center',}}>
          <View style={{width: 80}} >
            <Button onPress={showDatepicker} title="Start date picker"
            buttonStyle={styles.buttonStyle1}
            />
          </View>
          <View style={{width: 50}} >
          </View>
          <View style={{width: 80}} >
            <Button onPress={showTimepicker} title="Start time picker"
            buttonStyle={styles.buttonStyle1}
            />
          </View>
        </View>
      </View>
      <View style={{padding: 20}}>
        <Text style={styles.bigOrange1}>
        End time
        </Text>
        <Text style={styles.timeText}>
        {endDate.toDateString()}, {endDate.toLocaleTimeString()}
        </Text>
        <View style={{flexDirection: 'row', justifyContent: 'center',}}>
          <View style={{width: 80}} >
            <Button onPress={showDatepicker2} title={endButtonTitle1}
            buttonStyle={styles.buttonStyle1}
            />
          </View>
          <View style={{width: 50, }} >
          </View>
          <View style={{width: 80}} >
            <Button onPress={showTimepicker2} title={endButtonTitle2}
            buttonStyle={styles.buttonStyle1}
            />
          </View>
        </View>
      </View>
      <View style={{padding: 20}}>
        <Text style={styles.bigOrange1}>
        Event Tag
        </Text>
        <View style={{alignItems: 'center'}}>
          <Picker
            selectedValue={selectedValue}
            style={{ height: 50, width: 150 }}
            onValueChange={(itemValue, itemIndex) => setSelectedValue(itemValue)}
          >
            <Picker.Item label="Select Here" value="N/A" />
            <Picker.Item label="Study" value="Study" />
            <Picker.Item label="Work" value="Work" />
            <Picker.Item label="Relax" value="Relax" />
            <Picker.Item label="Sleep" value="Sleep" />
          </Picker>
        </View>
      </View>
      <View style={{alignItems: 'center'}}>
        <View style={{width: 150}} >
          <Button titleStyle={styles.buttonTitleStyle}
          buttonStyle={styles.buttonStyle2}
          onPress={() => {
            submitRecord(startDate, endDate, selectedValue);
          }}
          title="Submit"
          />
        </View>
      </View>
      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          timeZoneOffsetInMinutes={0}
          value={startDate}
          mode={mode}
          is24Hour={true}
          display="default"
          onChange={onStartChange}
        />
      )}
      {show2 && (
        <DateTimePicker
          testID="dateTimePicker2"
          timeZoneOffsetInMinutes={0}
          value={endDate}
          mode={mode2}
          is24Hour={true}
          display="default"
          onChange={onEndChange}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  bigText: {
    fontSize: 24,
    lineHeight: 35,
  },
  bigOrange1: {
    fontSize: 24,
    color: '#da4302',
    lineHeight: 35,
  },
  timeText: {
    textAlign: 'center',
    fontSize: 20,
    lineHeight: 50,
  },
  buttonStyle1: {
    backgroundColor: '#00818a',
  },
  buttonStyle2: {
    borderRadius: 30,
    backgroundColor: '#032535',
  },
  buttonTitleStyle: {
    fontSize: 26,
    color: 'white',
  },
});
