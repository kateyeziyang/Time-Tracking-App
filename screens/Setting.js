import * as React from 'react';
import { useState } from 'react';
import { Text, View, Dimensions, TextInput, Alert, StyleSheet } from 'react-native';
import { Button } from 'react-native-elements';
import { Picker } from '@react-native-community/picker';
import AsyncStorage from '@react-native-community/async-storage';
import { setNewPwd } from '../common/commonFunctions';

/**
 * App lock screen.
 * @return a page to set up your app lock.
*/
export function SettingScreen() {
    const [curPwd, onChangeText1] = React.useState('');
    const [newPwd, onChangeText2] = React.useState('');
    const alert1 = () => { Alert.alert("OK", "Successfully remove app lock."); }
    const alert2 = () => { Alert.alert("OK", "Successfully set up your app lock."); }
    const alert3 = () => { Alert.alert("Sorry", "Your current password is wrong."); }

  return(
    <View style={{flex: 1}}>
      <View elevation={5} style={{flex:0.12, flexDirection: 'row', alignItems: 'center', padding: 15, backgroundColor: '#f7be16'}}>
        <View style={{width: 130}} >
        </View>
        <View style={{width: 120, justifyContent: 'center', alignItems: 'center'}} >
          <Text style={styles.screenTitle}>
          App  Lock
          </Text>
        </View>
        <View style={{width: 20}} >
        </View>
        <View style={{width: 180, height: 40, justifyContent: 'center', alignItems: 'center'}}>
          <Button
          onPress={() => {
            setNewPwd(curPwd, newPwd, alert1, alert2, alert3);
            onChangeText1("");
            onChangeText2("");
          }}
          title="Save" type="outline"
          buttonStyle={styles.buttonStyle}/>
        </View>
      </View>
      <View style={{paddingLeft:20, paddingRight: 20}}>
        <View style={{padding:20}}>
          <Text style={styles.bigText}>
          Your current 4-digit PIN:
          </Text>
          <Text style={styles.smallText}>
          When you don't have one, leave it blank.
          </Text>
          <View style={{alignItems: 'center', padding: 20}}>
            <View style={{width: 80}}>
              <TextInput keyboardType="numeric" secureTextEntry={true} value={curPwd}
              style={{borderColor: '#ec9b3b', borderWidth: 2, backgroundColor: 'white', fontFamily:'monospace'}}
              maxLength={4} onChangeText={curPwd => onChangeText1(curPwd)}>
              </TextInput>
            </View>
          </View>
        </View>
        <View style={{paddingLeft:20, paddingRight: 20}}>
          <Text style={styles.bigText}>
          Your new 4-digit PIN:
          </Text>
          <Text style={styles.smallText}>
          Leave it blank if you don't want app lock.
          </Text>
          <View style={{alignItems: 'center', padding: 20}}>
            <View style={{width: 80}}>
              <TextInput keyboardType="numeric" secureTextEntry={true} value={newPwd}
              style={{borderColor: '#ec9b3b', borderWidth: 2, backgroundColor: 'white', fontFamily:'monospace'}}
              maxLength={4} onChangeText={newPwd => onChangeText2(newPwd)}>
              </TextInput>
            </View>
          </View>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  bigText: {
    fontSize: 18,
  },
  smallText: {
    fontSize: 14,
    fontStyle: "italic"
  },
  screenTitle: {
    fontSize: 25,
    color: 'white',
  },
  buttonStyle: {
    backgroundColor: 'white',
  },
});
