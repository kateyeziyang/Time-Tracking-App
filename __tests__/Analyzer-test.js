/**
 * @format
 */

import 'react-native';
import React from 'react';
import AsyncStorage from '@react-native-community/async-storage';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { AnalyzerScreen } from '../screens/Analyzer';

beforeEach(async () => {
  await AsyncStorage.clear();
});

it('renders correctly when there is no data available.', async() => {
  const tree = renderer
    .create(<AnalyzerScreen />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with pie chart when there is some data', async() => {
  const testValue = [{"color": "#363062", "count": 60, "legendFontColor": "#7F7F7F", "legendFontSize": 15, "name": "Study"}, {"color": "#4d4c7d", "count": 180, "legendFontColor": "#7F7F7F", "legendFontSize": 15, "name": "Work"}, {"color": "#827397", "count": 60, "legendFontColor": "#7F7F7F", "legendFontSize": 15, "name": "Relax"}, {"color": "#d8b9c3", "count": 120, "legendFontColor": "#7F7F7F", "legendFontSize": 15, "name": "Sleep"}];

  const tree = renderer
    .create(<AnalyzerScreen extra={testValue}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
