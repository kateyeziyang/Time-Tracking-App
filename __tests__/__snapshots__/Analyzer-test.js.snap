// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is no data available. 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 30,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5eb7b7",
              "fontSize": 25,
            }
          }
        >
          ❶
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          Pick a Time Range
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text>
        From
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 70,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#c2e8ce",
            "borderWidth": 2,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="04/13/20"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 30,
        }
      }
    >
      <Text>
        To
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 70,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#c2e8ce",
            "borderWidth": 2,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="04/19/20"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 90,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontStyle": "italic",
          }
        }
      >
        MM/DD/YY
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 30,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5eb7b7",
              "fontSize": 25,
            }
          }
        >
          ❷
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          Select Chart Type
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "width": 190,
          }
        }
      >
        <View
          style={
            Object {
              "height": 50,
              "width": 200,
            }
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Pie Chart",
                  "textColor": undefined,
                  "value": "Pie",
                },
                Object {
                  "label": "Stacked-Bar Chart",
                  "textColor": undefined,
                  "value": "Stacked",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "width": 20,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "width": 50,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fc7978",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with pie chart when there is some data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 30,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5eb7b7",
              "fontSize": 25,
            }
          }
        >
          ❶
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          Pick a Time Range
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text>
        From
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 70,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#c2e8ce",
            "borderWidth": 2,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="04/13/20"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 30,
        }
      }
    >
      <Text>
        To
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 70,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#c2e8ce",
            "borderWidth": 2,
            "height": 40,
          }
        }
        underlineColorAndroid="transparent"
        value="04/19/20"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 90,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontStyle": "italic",
          }
        }
      >
        MM/DD/YY
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 30,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5eb7b7",
              "fontSize": 25,
            }
          }
        >
          ❷
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
          Select Chart Type
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": 50,
            "width": 190,
          }
        }
      >
        <View
          style={
            Object {
              "height": 50,
              "width": 200,
            }
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Pie Chart",
                  "textColor": undefined,
                  "value": "Pie",
                },
                Object {
                  "label": "Stacked-Bar Chart",
                  "textColor": undefined,
                  "value": "Stacked",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "width": 20,
          }
        }
      />
      <View
        style={
          Object {
            "height": 30,
            "width": 50,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 14,
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fc7978",
                  "borderColor": "#2089dc",
                  "borderRadius": 14,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
